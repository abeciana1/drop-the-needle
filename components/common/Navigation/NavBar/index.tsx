import React, { useState } from 'react'
import Image from 'next/image'
import Link from 'next/link'
import { OnClickButton } from '@/components/common'
import useResponsiveness from '@/hooks/useResponsiveness'
import cx from 'classnames'
import { Squash as Hamburger } from 'hamburger-react'
import { signIn } from "next-auth/react"
import { NavBarI } from '@/interfaces'
import {
    AccountDD,
    NavItem
} from '@/components/common'

const NavBar = ({ status }: NavBarI) => {
    const [ isOpen, setOpen ] = useState(false)

    const {
        isMobile,
        isTablet,
        isDesktop
    } = useResponsiveness() || {}

    const handleGoogleSignin = async () => {
        await signIn('google', {callbackUrl: '/'})
    }

    return (
        <React.Fragment>
            <nav
                className={cx('z-50 pt-5 px-5 sm:px-10 lg:px-20', {
                    ['flex justify-between items-center ']: isDesktop,
                    ['bg-altWhite h-screen']: ((isMobile || isTablet) && isOpen)
                })}
            >
                <Link
                    href='/'
                    title='Navigate to homepage'
                >
                    <Image
                        src='/drop-the-needle-logo.webp'
                        height={150}
                        width={150}
                        alt='Drop The Needle logo'
                        priority
                        className='mx-auto'
                    />
                </Link>
                {(isOpen || isDesktop) &&
                    <React.Fragment>
                        <ul
                            data-testid={isOpen ? 'menu-open': 'menu-closed'}
                            className={cx('flex', {
                                ['flex-col pt-6 gap-6 pl-6']: isMobile || isTablet,
                                ['pt-0 gap-12']: isDesktop
                            })}
                        >
                            <NavItem
                                href=''
                                color='jaffa'
                                linkText='Home'
                            />
                            <NavItem
                                href='participate'
                                color='vermillion'
                                linkText='Participate'
                            />
                            <NavItem
                                href='listen'
                                color='altGreen'
                                linkText='Listen'
                            />
                        </ul>
                        {status === false &&
                            <li 
                                className={cx({
                                    ['pt-6 pl-6']: isMobile || isTablet 
                                })}
                            >
                                <OnClickButton
                                    onClick={handleGoogleSignin}
                                    text='Signin'
                                    bgColor='vermillion'
                                    ctaArrow={false}
                                />
                            </li>
                        }
                        {status &&
                            <div></div>
                        }
                    </React.Fragment>
                }
                {(isMobile || isTablet) &&
                    <div data-testid='hamburger-btn' className='absolute top-5 right-5'>
                        <Hamburger
                            toggled={isOpen}
                            toggle={setOpen}
                            color='#343434'
                        />
                    </div>
                }
            </nav>
        </React.Fragment>
    )
}

export default NavBar