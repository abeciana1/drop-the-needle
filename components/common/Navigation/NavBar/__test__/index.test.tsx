import { render, screen } from '@testing-library/react'
import NavBar from '..'
import { Context as ResponsiveContext } from 'react-responsive'


test('NavBar renders without issues', () => {
    render(<NavBar/>)
    const nav = screen.getByRole('navigation')
    expect(nav).toBeInTheDocument()
})

test('Navbar logo renders with anchor tag pointing to homepage', () => {
    render(<NavBar/>)
    const logo = screen.getByRole('img')
    expect(logo).toBeInTheDocument()
    expect(logo).toHaveAttribute('alt', 'Drop The Needle logo')
    expect(logo).toHaveAttribute('src', '/_next/image?url=%2Fdrop-the-needle-logo.webp&w=256&q=75')

    const link = screen.getByTitle(/navigate to homepage/i)
    expect(link).toBeInTheDocument()
    expect(link).toHaveAttribute('title', 'Navigate to homepage')
    expect(link).toHaveAttribute('href', '/')
})

test('[guest user && desktop] Signin link that looks like a button renders in NavBar', () => {
    const { container: desktop } = render(
        <ResponsiveContext.Provider value={{ width: 2000 }}>
            <NavBar/>
        </ResponsiveContext.Provider>
    )
    const signinLink = screen.getByRole('link', {
        name: /signin/i
    })

    expect(signinLink).toBeInTheDocument()
    expect(signinLink).toHaveAttribute('href', '/signin')
    expect(desktop).toMatchSnapshot()
})