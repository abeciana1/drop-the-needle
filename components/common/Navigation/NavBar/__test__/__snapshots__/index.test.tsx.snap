// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[guest user && desktop] Signin link that looks like a button renders in NavBar 1`] = `
<div>
  <nav
    class="z-50 pt-5 flex justify-between items-center "
  >
    <a
      href="/"
      title="Navigate to homepage"
    >
      <img
        alt="Drop The Needle logo"
        class="mx-auto"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="100"
        src="/_next/image?url=%2Fdrop-the-needle-logo.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fdrop-the-needle-logo.webp&w=128&q=75 1x, /_next/image?url=%2Fdrop-the-needle-logo.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="100"
      />
    </a>
    <ul
      class="flex pt-0 gap-12"
    >
      <li>
        <a
          class="flex flex-col font-medium text-2xl anim-text hover-underline-animation hover:text-jaffa jaffa text-jaffa after:bg-jaffa"
          href="/"
          title="Home page"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class="flex flex-col font-medium text-2xl anim-text hover-underline-animation hover:text-vermillion vermillion  text-vermillion after:bg-vermillion"
          href="/participate"
          title="Participate page"
        >
          Participate
        </a>
      </li>
      <li>
        <a
          class="flex flex-col font-medium text-2xl anim-text hover-underline-animation hover:text-altGreen altGreen text-altGreen after:bg-altGreen"
          href="/listen"
          title="Listen page"
        >
          Listen
        </a>
      </li>
    </ul>
    <div
      class=""
    >
      <a
        class="text-2xl font-medium px-3 py-1 rounded-lg text-altWhite bg-vermillion"
        href="/signin"
      >
        Signin
      </a>
    </div>
    <div
      class="absolute top-5 right-5"
      data-testid="hamburger-btn"
    >
      <div
        aria-expanded="false"
        class="hamburger-react"
        role="button"
        style="cursor: pointer; height: 48px; position: relative; transition: 0.4s cubic-bezier(0, 0, 0, 1); user-select: none; width: 48px; outline: none;"
        tabindex="0"
      >
        <div
          style="transition: 0.2s cubic-bezier(0, 0, 0, 1) 0.2s; transform: none;"
        >
          <div
            style="background: rgb(52, 52, 52); height: 3px; left: 8px; position: absolute; width: 32px; top: 13px; transition: 0.2s cubic-bezier(0, 0, 0, 1) 0s; transform: none;"
          />
        </div>
        <div
          style="transition: 0.2s cubic-bezier(0, 0, 0, 1); opacity: 1;"
        >
          <div
            style="background: rgb(52, 52, 52); height: 3px; left: 8px; position: absolute; width: 32px; top: 23px; transition: 0.2s cubic-bezier(0, 0, 0, 1);"
          />
        </div>
        <div
          style="transition: 0.2s cubic-bezier(0, 0, 0, 1) 0.2s; transform: none;"
        >
          <div
            style="background: rgb(52, 52, 52); height: 3px; left: 8px; position: absolute; width: 32px; top: 33px; transition: 0.2s cubic-bezier(0, 0, 0, 1) 0s; transform: none;"
          />
        </div>
      </div>
    </div>
  </nav>
</div>
`;
